Source/Public/Set-ADOPSServiceConnection.ps1

```diff
@@ -4,120 +4,65 @@ function Set-ADOPSServiceConnection {
-        
-            'ServicePrincipal' {
-                $authorization = [ordered]@{
-                    parameters = [ordered]@{
-                        tenantid            = $TenantId
-                        serviceprincipalid  = $ServicePrincipal.UserName
-                        authenticationType  = "spnKey"
-                        serviceprincipalkey = $ServicePrincipal.GetNetworkCredential().Password
-                    }
-                    scheme     = "ServicePrincipal"
-                }
-        
-                $data = [ordered]@{
-                    subscriptionId   = $SubscriptionId
-                    subscriptionName = $SubscriptionName
-                    environment      = "AzureCloud"
-                    scopeLevel       = "Subscription"
-                    creationMode     = "Manual"
-                }
-            }
-    
-            'ManagedServiceIdentity' {
-                $authorization = [ordered]@{
-                    parameters = [ordered]@{
-                        tenantid = $TenantId
-                    }
-                    scheme     = "ManagedServiceIdentity"
-                }
-            }
-        }
+        [Parameter(Mandatory, ParameterSetName = 'WorkloadIdentityFederation')]
+        [switch]$WorkloadIdentityFederation,
 
-        # Create body for the API call
-        $Body = [ordered]@{
-            authorization                    = $authorization
-            data                             = $data
-            description                      = "$Description"
-            id                               = $ServiceConnectionId
-            isReady                          = $true
-            isShared                         = $false
-            name                             = ($SubscriptionName -replace " ")
-            serviceEndpointProjectReferences = @(
-                [ordered]@{
-                    projectReference = [ordered]@{
-                        id   = $ProjectInfo.Id
-                        name = $Project
-                    }
-                    name             = $ConnectionName
-                }
-            )
-            type                             = "AzureRM"
-            url                              = "https://management.azure.com/"
-        } | ConvertTo-Json -Depth 10
-    
-        if ($PSBoundParameters.ContainsKey('EndpointOperation')) {
-            $URI = "https://dev.azure.com/$Organization/_apis/serviceendpoint/endpoints/$ServiceEndpointId`?operation=$EndpointOperation`&api-version=7.1-preview.4"
-        }
-        else {
-            $URI = "https://dev.azure.com/$Organization/_apis/serviceendpoint/endpoints/$ServiceEndpointId`?api-version=7.1-preview.4"
-        }
-        
-        $InvokeSplat = @{
-            Uri          = $URI
-            Method       = "PUT"
-            Body         = $Body
-        }
+        [Parameter(Mandatory, ParameterSetName = 'WorkloadIdentityFederation')]
```

Comment:
Is there any possibility to _not_ have this parameter as Mandatory? As it is a new parameter adding it as mandatory would mean a major break and a major version bump of the release. Is it possible to create a WIF Service connection without a scope?

Reply:
`Set-ADOPSServiceConnection` did not support `WorkloadIdentityFederation` parameter set, so this would be a "new" feature.

I am happy to remove this parameter set for a future date. Let me know what your thoughts are.